ws-server/src/server.rs

```diff
@@ -74,15 +74,15 @@ impl Server {
 	/// Start responding to connections requests. This will block current thread until the server is stopped.
 	pub async fn start(self) {
 		let mut incoming = TcpListenerStream::new(self.listener);
-		let methods = Arc::new(self.methods);
+		let methods = self.methods;
 		let cfg = self.cfg;
 		let mut id = 0;
 
 		while let Some(socket) = incoming.next().await {
 			if let Ok(socket) = socket {
 				socket.set_nodelay(true).unwrap_or_else(|e| panic!("Could not set NODELAY on socket: {:?}", e));
 
-				if Arc::strong_count(&methods) > self.cfg.max_connections as usize {
+				if methods.ref_count() > self.cfg.max_connections as usize {
```

Comment:
Yeah, good point, will probably need to track this separately (maybe spawn a dummy `Arc<()>`?). I'm also happy about removing `ref_count` from the API since that's a bleeding abstraction.

Reply:
yeah, we could have a separate `AtomicCounter` in the WsServer to keep track of the number of concurrent connections